waku/v2/node/storage/peer/waku_peer_storage.nim

```diff
@@ -47,9 +49,12 @@ proc encode*(storedInfo: StoredInfo): ProtoBuffer =
   for proto in storedInfo.protos.items:
     pb.write(3, proto)
   
-  pb.write(4, storedInfo.publicKey)
+  try:
+    pb.write(4, storedInfo.publicKey)
+  except ResultError[CryptoError] as e:
```

Comment:
I am asking this for my own understanding, I thought we should use `Result` when multiple failure paths exist, is it the case here? It seems there is only one failure path

Reply:
I have a general question regarding the use of `Result` but related to the files in this PR, if you check line https://github.com/status-im/nim-waku/blob/37c5a6eb222921375a2f9bcedc9133b2c4b1b705/waku/v2/node/storage/peer/waku_peer_storage.nim#L23, the output is defined as `ProtoResult[T]` however, this proc never results in an error, why do we need the Result type? I am going over the waku protocol modules and was wondering whether we should remove such `Result` usage or not?

Reply:
Yeah, I don't have strong feelings about this, but I think `Result` is also prudent to use where we can imagine multiple failure paths in future. Encoding/decoding is a good example, because this can often fail in multiple ways. I'd rather elevate more return types to `Result` than remove `Result` for functions where we don't return `err` yet. There's also the consistency of other protobuf `init*` functions returning `ProtoResult`, and we probably want to maintain this pattern. Another reason to use `Result` above `Opt`, for example, is that you can be explicit about what caused the error.

Reply:
Thanks for the reply! 
What do you refer to by `Opt`, is it `Option`?

Reply:
Indeed!